Add sugessted approvers for MergeRequestApprovals request, AssigneeID…

h0ckeypl4yer opened the pull request:
Add sugessted approvers for MergeRequestApprovals request, AssigneeIDs for Merge Requests

svanharmelen commented on merge_request_approvals.go:
```diff
@@ -33,6 +33,7 @@ type MergeRequestApprovals struct {
 	ApprovedBy           []*MergeRequestApproverUser  `json:"approved_by"`
 	Approvers            []*MergeRequestApproverUser  `json:"approvers"`
 	ApproverGroups       []*MergeRequestApproverGroup `json:"approver_groups"`
+	SuggestedApprovers   []*ApproverUser              `json:"suggested_approvers"`
```
I cannot find this in the docs, can you point me to a location that shows this is indeed a valid field?

h0ckeypl4yer replied:
There is no information about this parameter in docs. But it works. Version GitLab Enterprise Edition 12.2.5-ee

svanharmelen replied:
I need to see some reference to this in order to be able to judge if this is the best way to add this field. A link to the definition in the Gitlab source code would also be fine.

itscaro replied:
Cf. https://gitlab.com/gitlab-org/gitlab/blob/master/ee/lib/ee/api/entities.rb#L425-428

svanharmelen commented on merge_requests.go:
```diff
@@ -476,6 +476,7 @@ type CreateMergeRequestOptions struct {
 	TargetBranch       *string `url:"target_branch,omitempty" json:"target_branch,omitempty"`
 	Labels             Labels  `url:"labels,comma,omitempty" json:"labels,omitempty"`
 	AssigneeID         *int    `url:"assignee_id,omitempty" json:"assignee_id,omitempty"`
+	AssigneeIDs        []*int  `url:"assignee_ids,omitempty" json:"assignee_ids,omitempty"`
```
```suggestion
	AssigneeIDs        []int  `url:"assignee_ids,omitempty" json:"assignee_ids,omitempty"`
```

svanharmelen commented on merge_requests.go:
```diff
@@ -518,6 +519,7 @@ type UpdateMergeRequestOptions struct {
 	Description        *string `url:"description,omitempty" json:"description,omitempty"`
 	TargetBranch       *string `url:"target_branch,omitempty" json:"target_branch,omitempty"`
 	AssigneeID         *int    `url:"assignee_id,omitempty" json:"assignee_id,omitempty"`
+	AssigneeIDs        []*int  `url:"assignee_ids,omitempty" json:"assignee_ids,omitempty"`
```
```suggestion
	AssigneeIDs        []int  `url:"assignee_ids,omitempty" json:"assignee_ids,omitempty"`
```

svanharmelen commented on merge_request_approvals.go:
```diff
@@ -60,7 +61,7 @@ type MergeRequestApproverGroup struct {
 // https://docs.gitlab.com/ee/api/merge_request_approvals.html#project-level-mr-approvals
 type MergeRequestApproverUser struct {
-	User struct {
+	ApproverUser struct {
```
This will break code that uses `MergeRequestApproverUser` as now the expected (`User`) field name changed (to `ApprovedUser`). So please revert this change.

svanharmelen commented:
Thanks for updating the PR. Can you please rebase the PR and as a last verification (sorry for all the questions, but I don't have an instance to test against myself at the moment) provide a (censored) API response that contains this field? Thanks!!

h0ckeypl4yer commented:
> Thanks for updating the PR. Can you please rebase the PR and as a last verification (sorry for all the questions, but I don't have an instance to test against myself at the moment) provide a (censored) API response that contains this field?
> 
> _EDIT:_ I just merged PR #667 which adds a `BasicUser` struct which could (should) be used here as well. So could you please update hat as well? Thanks!

I updated the merge request and added a screenshot from PostMan. I hope this is enough.
![Снимок экрана 2019-09-24 в 12 33 40](https://user-images.githubusercontent.com/7531693/65501447-b3eda700-dec9-11e9-9eff-baceb77bf717.png)
